Sources/LeafKit/LeafLexer/LeafToken.swift

```diff
@@ -300,6 +336,37 @@ internal extension Array where Element == ParameterDeclaration {
     func describe(_ joinBy: String = " ") -> String {
         return self.map {$0.short }.joined(separator: joinBy)
     }
+    
+    func imports() -> Set<String> {
+        var result = Set<String>()
+        self.forEach { result.formUnion($0.imports()) }
```

Comment:
NitPick: `return reduce(Set<String>()) { $0.formUnion($1.imports()) }`
Or even: `Set(lazy.flatMap { $0.imports() })`

Reply:
Leaving as-is for clarity; `reduce` won't compile as you wrote it and while I'd originally tried to use it anyway, it gets too wonky looking IMO. Just not a fan of lazy in this usage since it masks what's happening

Reply:
Ah, should have been `reduce(into: Set<String>()) { ... }`, sorry. The `into:` version passes a mutating parameter into the closure.

Same goes for the other similar cases I mentioned.